### DISCUSSION TITLE
TTL with spring-cloud-aws-starter-dynamodb DynamoDbTemplate

### DISCUSSION THREAD

**Discussion by huehnerlady** [awspring/spring-cloud-aws#874] [Q&A] [answered]
Hi,

I am currently migrating our service to spring-boot 3 and saw the changes made here. I saw there is a DynamoDbTemplate and the usage looks quite nice an simple in comparison to our use of the `Table` object before, but so far I could not find out how to set a ttl for the items in a table. In the properties provided all I can see is a connection-ttl.

Can anybody maybe assist with an example?

**Comment by dominik-kovacs** [comment, marked as answer]
Lately I was thinking about this as well. TTL of an item is based on an items attribute of your choice.

For example you can create an attribute called `expirationDate` with data type long (must be epoch seconds) and populate it yourself. Then in the aws console you tell dynamodb that `expirationDate` is your ttl attribute and it will take care of the rest.
